torch/_dynamo/variables/builder.py

```diff
@@ -1710,7 +1712,17 @@ def update_dim2constraint(dim, constraint_range, debug_name):
         constraint = dim2constraint.get(i)
         if constraint is None:
             if marked_dynamic and not config.allow_ignore_mark_dynamic:
-                constraint_dim = RelaxedUnspecConstraint(warn_only=False)
+                # _dynamo_dynamic_indices will be of the form (index, min, max)
+                index_ranges = [ir for ir in getattr(e, "_dynamo_dynamic_indices") if ir[0] == i]
+                min_val = index_ranges[0][1]
+                max_val = index_ranges[0][2]
+                if min_val == 2 and not max_val:
+                    constraint_dim = RelaxedUnspecConstraint(warn_only=False)
+                else:
+                    constraint_dim = StrictMinMaxConstraint(
```

Comment:
I'm a little concerned about this going from Relaxed to Strict, the main question is whether or not export requires there are NO other constraints, or if it's just testing the value range. If it's just value range that should be alright.

Reply:
As per my understanding ([of this snippet](https://github.com/pytorch/pytorch/blob/main/torch/export/dynamic_shapes.py#L124-L147) and this file), it seems like it's just testing the value range, but I'm not completely sure